### PR TITLE
use import-fresh

### PR DESCRIPTION
**Pull request by olsonpm**
new direction from #157

The new pull request was to get rid of the messy diffs and misleading branch name/pr name/etc.

### CODE REVIEW

File: src/importFreshWrapper.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+//
+// this is one solution to make importFresh spy'able
+// see: https://github.com/sinonjs/sinon/issues/664
+//
+
+const importFresh = require('import-fresh');
+
+module.exports = { importFresh };
```
Comment (davidtheclark) [inline src/importFreshWrapper.js]: With Jest we can mock modules with [`jest.mock`](https://jestjs.io/docs/en/jest-object#jestmockmodulename-factory-options), so I don't think this wrapper is necessary.

Comment (olsonpm) [reply src/importFreshWrapper.js]: Ugh - I'm sorry but I'm having trouble figuring out how to organize the mock code and would appreciate some direction.  I tried adding

```js
const mockedRequireFresh = jest.fn(() => ({ foundJs: true }));
jest.mock('import-fresh', mockedRequireFresh);

const cosmiconfig = require('../src');
```

but then something happens to the mock by the time the test runs where it returns undefined.  From what I read `jest.restoreAllMocks` shouldn't be at fault and that proved true when I commented that line out.

I haven't used jest before.  It has a large api surface and seems to be doing some magical things (e.g. its own `require` took me a bit to figure out).

**Comment by sudo-suhas** [comment]
Diff which uses `jest.mock`:
```diff
diff --git a/src/loaders.js b/src/loaders.js
index bccf4d4..c156c97 100644
--- a/src/loaders.js
+++ b/src/loaders.js
@@ -1,13 +1,12 @@
 // @flow
 'use strict';
 
+const importFresh = require('import-fresh');
 const parseJson = require('parse-json');
 const yaml = require('js-yaml');
 
-const wrapper = require('./importFreshWrapper');
-
 function loadJs(filepath: string): Object {
-  const result = wrapper.importFresh(filepath);
+  const result = importFresh(filepath);
   return result;
 }
 
diff --git a/test/caches.test.js b/test/caches.test.js
index 90a1f0d..6500600 100644
--- a/test/caches.test.js
+++ b/test/caches.test.js
@@ -1,9 +1,12 @@
 'use strict';
 
 const fs = require('fs');
+const importFreshMock = require('import-fresh');
 const util = require('./util');
 const cosmiconfig = require('../src');
-const wrapper = require('../src/importFreshWrapper');
+
+// mocks are hoisted
+jest.mock('import-fresh')
 
 const temp = new util.TempDir();
 
@@ -562,29 +565,31 @@ describe('ensure import-fresh is called when loading a js file', () => {
   const tempFileName = 'a/b/c/d/.foorc.js';
   const loadPath = temp.absolutePath(tempFileName);
 
-  const checkResult = (importFreshSpy, result) => {
-    expect(importFreshSpy).toHaveBeenCalledTimes(1);
-    expect(importFreshSpy).toHaveBeenCalledWith(loadPath);
+  beforeEach(() => {
+    importFreshMock.mockImplementation(() => ({ foundJs: true }))
+  })
+
+  const checkResult = result => {
+    expect(importFreshMock).toHaveBeenCalledTimes(1);
+    expect(importFreshMock).toHaveBeenCalledWith(loadPath);
 
     expect(result.config).toEqual({ foundJs: true });
   };
 
   test('async', () => {
-    const importFreshSpy = jest.spyOn(wrapper, 'importFresh');
     const explorer = cosmiconfig('foo');
     temp.createFile(tempFileName, 'module.exports = { foundJs: true };');
 
     return explorer.load(loadPath).then(result => {
-      checkResult(importFreshSpy, result);
+      checkResult(result);
     });
   });
 
   test('sync', () => {
-    const importFreshSpy = jest.spyOn(wrapper, 'importFresh');
     const explorer = cosmiconfig('foo');
     temp.createFile(tempFileName, 'module.exports = { foundJs: true };');
 
     const result = explorer.loadSync(loadPath);
-    checkResult(importFreshSpy, result);
+    checkResult(result);
   });
 });

```

**Comment by olsonpm** [comment]
hey thanks so much.  I'll have that committed tonight or tomorrow.

**Comment by sudo-suhas** [comment]
No problem. I too have wrestled with `jest.mock` in some other projects.

**Pull request merged**